extensions/amp-social-share/0.2/social-share.js

```diff
@@ -30,63 +32,196 @@ const DEFAULT_HEIGHT = 44;
  * @param {!JsonObject} props
  * @return {PreactDef.Renderable}
  */
-export function SocialShare(props) {
+export function SocialShare2(props) {
+  const name = 'SocialShare2';
+  useResourcesNotify();
+
+  const {
+    typeConfig,
+    baseEndpoint,
+    checkedWidth,
+    checkedHeight,
+    obsoleteType,
+  } = checkProps_(props, name);
+  if (obsoleteType) {
+    return;
+  }
+  const finalEndpoint = createEndpoint_(typeConfig, baseEndpoint, props);
+
   /**
-   * Handle key presses on the element.
-   * @param {!Event} event
    * @private
+   * @param {!JsonObject} props
+   * @param {?string} name
+   * @return {!JsonObject}
    */
-  function handleKeyPress(event) {
-    const {key} = event;
-    if (key == Keys.SPACE || key == Keys.ENTER) {
-      event.preventDefault();
-      handleActivation();
+  function checkProps_(props, name) {
+    const {type, shareEndpoint, params, bindings, width, height} = props;
+
+    // Verify type is valid and cannot contain spaces
+    if (type === undefined) {
+      throwError_(`The type attribute is required. ${name}`);
```

Comment:
Why not just do `throw new Error()` here? If we actually throw errors, it'd be better to be explicit here, since the compiler understands that the function exits with the `throw` statement.

But, I'm not yet sold we should be throwing errors here. I think they are somewhat unidiomatic in React development. But let's do it for now, until we figure out a better way.

Reply:
Yes, I wrapped the actual throwing of an error inside it's own function in case we want to change it throughout the component, since I am not sure what is the idiomatic way to throw an error in React and specifically for bento.

Sure I can make it explicit here.

Reply:
We just need to be consistent here. If it's a method call that may or may not throw in different cases, we have to propagate `null/undefined` values, and then skip rendering by returning null. If we `throw` errors and don't want to deal with this yet, we should make `throw` explicit to compiler and the only way to do it is to actually have `throw ...` at the call site. I recommend we start by throwing actual errors explicitly - it's less effort for us for now and we'd likely have to pass through the errors again at some point.

Reply:
Makes sense, let's throw the error explicitly.